src/promptflow/promptflow/executor/flow_executor.py

```diff
@@ -711,12 +711,31 @@ def exec_line(
                 validate_inputs=validate_inputs,
                 allow_generator_output=allow_generator_output,
             )
-        self._save_image_from_output(line_result.output, self._working_dir)
         #  Return line result with index
         if index is not None and isinstance(line_result.output, dict):
             line_result.output[LINE_NUMBER_KEY] = index
         return line_result
 
+    def _exec_line_with_dir(
+        self,
+        inputs: Mapping[str, Any],
+        index: Optional[int] = None,
+        run_id: Optional[str] = None,
+        variant_id: str = "",
+        validate_inputs: bool = True,
+        node_concurrency=DEFAULT_CONCURRENCY_FLOW,
+        allow_generator_output: bool = False,
+        input_dir: Optional[Path] = None,
+        output_dir: Optional[Path] = None,
+    ):
+        inputs = self._process_images_from_inputs(self._flow.inputs, inputs, input_dir)
+        line_result = self.exec_line(
+            inputs, index, run_id, variant_id, validate_inputs, node_concurrency, allow_generator_output
+        )
+        if output_dir:
+            self._persist_images_from_output(line_result.output, output_dir)
```

Comment:
Here we need to write: `line_result = self._persist_images_from_output(line_result.output, output_dir)` to make sure it is clear that line_result is updated.

Reply:
Refined.